src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -141,22 +135,9 @@ class AuthScreens extends React.Component {
         // Fetch some data we need on initialization
         NameValuePair.get(CONST.NVP.PRIORITY_MODE, ONYXKEYS.NVP_PRIORITY_MODE, 'default');
         NameValuePair.get(CONST.NVP.IS_FIRST_TIME_NEW_EXPENSIFY_USER, ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER, true);
-
-        API.Get({
-            returnValueList: 'nameValuePairs',
-            nvpNames: ONYXKEYS.NVP_PREFERRED_LOCALE,
-        }).then((response) => {
-            const preferredLocale = lodashGet(response, ['nameValuePairs', 'preferredLocale'], CONST.DEFAULT_LOCALE);
-            if ((currentPreferredLocale !== CONST.DEFAULT_LOCALE) && (preferredLocale !== currentPreferredLocale)) {
-                setLocale(currentPreferredLocale);
-            } else {
-                Onyx.set(ONYXKEYS.NVP_PREFERRED_LOCALE, preferredLocale);
-            }
-        });
-
         PersonalDetails.fetchPersonalDetails();
         User.getUserDetails();
-        User.getBetas();
+        User.fetchBetasAndSetPreferredLocale(this.props.preferredLocale);
```

Comment:
I don't get why we are setting the preferredLocale here, the user is logging in, their locale can't change in this flow.

Reply:
Could you please clarify this for me? I didn't get it. 

I think, if the user does not have access to betas then we don't want the language to be changed from En as the locale is a beta feature.

Reply:
> Could you please clarify this for me? I didn't get it.

I don't get why we need to set a locale here nor where the value from the locale will come. The only place where the user can set the locale is from the preferences page, no?

> I think, if the user does not have access to betas then we don't want the language to be changed from En as the locale is a beta feature.

The way I see it, regardless of the beta, the locale should be what's in preferred locale or english if nothing is set in preferred locale.

Reply:
Here, the locale is picked from Onyx if the user has set a locale from the login page picker.
![image](https://user-images.githubusercontent.com/24370807/139670768-04e26190-82b4-47a6-958c-94019c2a8890.png)

I am following the existing code. Just added a check for beta so that we can break the loop if the user does not have localization beta access.

> The way I see it, regardless of the beta, the locale should be what's in preferred locale or english if nothing is set in the preferred locale.

Yup, it does but with one exception that it also picks the locale preference from the login page.

Reply:
If we kill the login page language selection , would that mean we can remove this?

Reply:
Then we will have to remove the extra code.